src/workflow/restore_from_mnemonic.c

```diff
@@ -113,27 +113,28 @@ static bool _get_mnemonic(char* mnemonic_out)
     workflow_status_create(num_words_success_msg, true);
 
     for (uint8_t word_idx = 0; word_idx < num_words; word_idx++) {
-        char word[WORKFLOW_TRINARY_INPUT_MAX_WORD_LENGTH] = {0};
+        char word[WORKFLOW_TRINARY_INPUT_MAX_WORD_LENGTH + 1] = {0};
         char title[50] = {0};
         _set_title(word_idx, title, sizeof(title));
         if (!workflow_trinary_input_wordlist(
                 title, (const char* const*)wordlist, BIP39_WORDLIST_LEN, word)) {
             return false;
         }
         if (word_idx != 0) {
-            strcat(mnemonic_out, " ");
+            strcat(mnemonic_out, " "); // NOLINT (gcc and clang cannot agree on best practice here)
         }
-        strcat(mnemonic_out, word);
+        strncat(mnemonic_out, word, WORKFLOW_TRINARY_INPUT_MAX_WORD_LENGTH);
     }
     return true;
 }
 
 bool workflow_restore_from_mnemonic(void)
 {
-    // same as: (MAX_WORD_LENGTH-1) * MAX_WORDS + (MAX_WORDS-1) + 1
+    // same as: MAX_WORD_LENGTH * MAX_WORDS + MAX_WORDS - 1 + 1
```

Comment:
please keep the parens around `(MAX_WORDS - 1)` for better grouping with the next line. `-1 + 1` just looks silly :P

Reply:
ha, you commented on the comment but I think there is a bug in the next chunk. 

```
(WORKFLOW_TRINARY_INPUT_MAX_WORD_LENGTH + 1) * WORKFLOW_RESTORE_FROM_MNEMONIC_MAX_WORDS + 1
```

should be

```
(WORKFLOW_TRINARY_INPUT_MAX_WORD_LENGTH + 1) * WORKFLOW_RESTORE_FROM_MNEMONIC_MAX_WORDS
```

right?

Reply:
@benma After you have double checked this I'll merge this branch

Reply:
`(WORKFLOW_TRINARY_INPUT_MAX_WORD_LENGTH + 1) * WORKFLOW_RESTORE_FROM_MNEMONIC_MAX_WORDS` looks correct to me. can't  recall seeing anything else before, but the previous commit is gone now.